nimble_build_system/cad/shelf.py

```diff
@@ -96,15 +106,34 @@ class Shelf():
     _variant = None
     _device = None
     _device_model = None
+    _shelf_model = None
+    _shelf_assembly_model = None
+    _exploded_shelf_assembly_model = None
     _assembled_shelf = None
+    _device_depth_axis = None  # Can be items like "-X", "X", "-Y", etc
+    _device_offset = (0, 0, 0)  # Offsets to put the device for the correct assembly position
+    _device_explode_translation = (0, 0, 0)  # Where to move the device to during an explode
+    _hole_locations = None  # List of hole locations for the device
+    _fasteners = []  # List of screw positions for the device
     _unit_width = 6  # 6 or 10 inch rack
+    _render_options = None
+
+    # Hole location parameters
+    _screw_dist_x = None
+    _screw_dist_y = None
+    _dist_to_front = None
+    _offset_x = None
```

Comment:
I can imagine that for some of the shelves we will need different variables because most don't have screws coming top-down. some come in from the sides. We may want to move these attributes to the specific classes.

Reply:
> because most don't have screws coming top-down. some come in from the sides. We may want to move these attributes to the specific classes.

There is a parameter for this. You set the axis of alignment for the screw to get it in the right orientation.